log/formatters.go

```diff
@@ -23,27 +26,77 @@ type jsonFormatter struct {
 	timestampColumn string
 }
 
+// Defining a log line type so that we can use a much faster JSON marshall-ing package
+type logData struct {
+	level        string
+	timestamp    string
+	timestampKey string
+	message      string
+	params       []*Field
+}
+
+// Implementing Marshaler
+func (m *logData) MarshalJSONObject(enc *gojay.Encoder) {
+	enc.StringKey("level", m.level)
+	enc.StringKey(m.timestampKey, m.timestamp)
+	enc.StringKey("message", m.message)
+
+	for _, v := range m.params {
+		switch vv := v.Value().(type) {
+		case string:
+			enc.StringKey(v.Key, vv)
+		case []string:
+			enc.AddSliceStringKey(v.Key, vv)
+		case int:
+			enc.IntKey(v.Key, vv)
+		case int32:
+			enc.Int32Key(v.Key, vv)
+		case int64:
+			enc.Int64Key(v.Key, vv)
+		case uint16:
+			uIntVal, _ := v.Value().(uint16)
```

Comment:
why the double cast here? vv should already have the uint16 value at this case. also, we do not support uint16 in our params (the Field struct), so is this really required here?

Reply:
That is strange. We support uint which is not supported by Gojay directly. Only uint64 and uint32 
This must have changed in our refactoring. I'll fix it now.